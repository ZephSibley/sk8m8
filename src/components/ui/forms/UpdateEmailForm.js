import React from 'react';
import { Formik } from 'formik';
import * as Yup from 'yup';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

import formStyles from '../../../styles/forms';


const UpdateEmailForm = () => {
    const formClasses = formStyles();

    const validationSchema = Yup.object().shape({
        email: Yup.string()
            .email("Please enter a valid email address")
            .required("Please enter your email address"),
        confirm_email: Yup.string()
            .oneOf([Yup.ref('email'), null], 'Emails must match'),
    })
    
    return (
        <Formik
            initialValues={{
                email: '',
                confirm_email: '',
                username: '',
                password: '',
                confirm_password: '',
            }}
            validationSchema={validationSchema}
            onSubmit={(values, { setSubmitting }) =>{
                console.log(values)
                setSubmitting(true);
            }}
            
            render={({
                values,
                errors,
                touched,
                handleChange,
                handleBlur,
                handleSubmit,
                isSubmitting,
                isValid,
            }) => (
                <form
                    onSubmit={handleSubmit}
                    className={formClasses.form}
                >
                    <TextField
                        name='email'
                        type='email'
                        label='email'
                        onChange={handleChange}
                        onBlur={handleBlur}
                        value={values.email}
                        margin='normal'
                        error={errors.email && touched.email}
                        helperText={
                            errors.email && touched.email ?
                            errors.email : ''
                        }
                    />
                    <TextField
                        name='confirm_email'
                        type='email'
                        label='confirm email'
                        onChange={handleChange}
                        onBlur={handleBlur}
                        value={values.confirm_email}
                        margin='normal'
                        error={errors.confirm_email && touched.confirm_email}
                        helperText={
                            errors.confirm_email && touched.confirm_email ?
                            errors.confirm_email : ''
                        }
                    />
                    <Button
                        variant="outlined"
                        color="primary"
                        component="button"
                        type="submit"
                        margin='normal'
                        className={formClasses.submit}
                        disabled={isSubmitting || !isValid}
                    >
                        Update Email
                    </Button>
                </form>
            )}
        />
    )
}

export default UpdateEmailForm;