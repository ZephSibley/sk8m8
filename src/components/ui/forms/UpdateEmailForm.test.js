import React from 'react';
import renderer from 'react-test-renderer';
import UpdateEmailForm from './UpdateEmailForm';

test('Basic Rendering', () => {
    const component = renderer.create(
        <UpdateEmailForm />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})