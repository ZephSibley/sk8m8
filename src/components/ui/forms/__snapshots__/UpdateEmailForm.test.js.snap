// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<form
  className="makeStyles-form-1"
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      email
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="email"
        value=""
      />
    </div>
    
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      confirm email
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="confirm_email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="email"
        value=""
      />
    </div>
    
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root makeStyles-submit-3 MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
    disabled={true}
    margin="normal"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Update Email
    </span>
  </button>
</form>
`;
